did_iota/src/producer/produce.rs

```diff
@@ -0,0 +1,179 @@
+use identity_iota::{
+    core::ToJson,
+    document::CoreDocument,
+    iota::{IotaDID, NetworkName},
+    storage::KeyId,
+};
+use log::info;
+use shared::{error::ProducerError, JwkStorageWrapper};
+
+use crate::producer::resolve::resolve;
+
+pub enum IotaMethod {
+    Testnet,
+    Shimmer,
+    Mainnet,
+}
+
+/// Note: Producing a DID document on an IOTA network involves publishing it to the network.
+pub async fn produce_did_iota(
+    storage: &JwkStorageWrapper,
+    key_id: &KeyId,
+    iota_method: IotaMethod,
+    managed_did: IotaDID,     // TODO(selv): see README.md
+    managed_fragment: String, // TODO(selv): see README.md
+) -> Result<CoreDocument, ProducerError> {
+    let stronghold_storage = match storage {
+        JwkStorageWrapper::Stronghold(stronghold_storage) => stronghold_storage,
+        JwkStorageWrapper::PKCS11 => todo!(),
+    };
+
+    // Sanity check: Does the key exist in storage?
+    let public_key_jwk = stronghold_storage.get_public_key(key_id).await.unwrap();
+
+    let _ = match iota_method {
+        IotaMethod::Testnet => {
+            info!(
+                "Producing did:iota:rms (Testnet) for key_id=[{:?}] ...",
+                key_id.as_str()
+            );
+            NetworkName::try_from("rms").unwrap()
```

Comment:
`[resolved]`
You could consider to use `.expect("some message")` here to 'mark' this type as code as code that does not need any error management (because you know `did_iota/src/producer/produce.rs` will never(?) fail). 

unwraps have a ambiguous meaning in production code imo, because it is either a 'forgotten to implement error handling' situation, or 'this will never fail' situation. 

Personally I like to approach `unwraps` and `expects` like this:
* use unwraps only while your code is still being developed. Removed them once your work is done and replace them with some kind of error management.
* for those `unwraps` that cannot possibly fail, replace them for an `expect("some message")`

This way, there will never be an `unwrap` in production code

Reply:
That's a very good approach! :) I think I will enable some clippy rule to lint against `unwrap()` altogether. 👍